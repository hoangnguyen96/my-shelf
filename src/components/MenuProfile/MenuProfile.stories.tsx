import { Meta, StoryFn } from "@storybook/react";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "@app/themes";
import MenuProfile from ".";

const meta: Meta<typeof MenuProfile> = {
  component: MenuProfile,
  decorators: [
    (Story: StoryFn) => (
      <ChakraProvider theme={theme}>
        <Story />
      </ChakraProvider>
    ),
  ],
};

export default meta;

const Template: StoryFn<typeof MenuProfile> = (args) => <MenuProfile />;

export const Default = Template.bind({});
Default.args = {};
