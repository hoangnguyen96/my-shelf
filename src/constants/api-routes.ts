export const API_ROUTES = {
  USER: "/user",
  BOOKS: "/books",
};
