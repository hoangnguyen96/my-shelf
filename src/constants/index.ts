export * from "./input";
export * from "./navbar";
export * from "./routes";
export * from "./api-routes";
export * from "./validate";
export * from "./message";
export * from "./filter";
export * from "./time";
