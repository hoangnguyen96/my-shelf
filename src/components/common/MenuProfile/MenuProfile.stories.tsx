import { Meta, StoryFn } from "@storybook/react";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "@app/themes";
import MenuProfile from ".";

const meta: Meta<typeof MenuProfile> = {
  component: MenuProfile,
  decorators: [
    (Story: StoryFn) => (
      <ChakraProvider theme={theme}>
        <Story />
      </ChakraProvider>
    ),
  ],
};

export default meta;

const Template: StoryFn<typeof MenuProfile> = (args) => (
  <MenuProfile
    user={{
      user: {
        id: "123",
        name: "John Doe",
        email: "johndoe@example.com",
        image: "https://example.com/johndoe.jpg",
        isAdmin: true,
      },
      expires: "2024-12-31T23:59:59.999Z",
    }}
    onRedirectFavorites={() => {}}
    onRedirectProfile={() => {}}
    onLogout={() => {}}
  />
);

export const Default = Template.bind({});
Default.args = {};
