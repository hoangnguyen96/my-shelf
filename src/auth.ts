import NextAuth from "next-auth";
import { authConfig } from "./auth.config";
import Credentials from "next-auth/providers/credentials";
import { z } from "zod";
import bcrypt from "bcryptjs";
import { MESSAGES } from "./constants";
import { getUserByEmail } from "@app/features/dashboard/actions";
import { User } from "./models";

export const { handlers, auth, signIn, signOut } = NextAuth({
  ...authConfig,
  secret: process.env.AUTH_SECRET,
  providers: [
    Credentials({
      name: "Credentials",
      credentials: {
        email: { label: "Email", type: "text" },
        password: { label: "Password", type: "password" },
      },
      async authorize(credentials) {
        if (!credentials) {
          return null;
        }

        const parsedCredentials = z
          .object({ email: z.string().email(), password: z.string().min(6) })
          .safeParse(credentials);

        if (parsedCredentials.success) {
          const { email, password } = parsedCredentials.data;
          const user = (await getUserByEmail(email)) as User;

          if (!user || !user.password) {
            throw new Error(MESSAGES.LOGIN_NOTFOUND);
          }

          const passwordsMatch = await bcrypt.compare(password, user.password);

          if (passwordsMatch)
            return {
              id: user.userId,
              name: user.username,
              isAdmin: user.isAdmin,
              image: user.avatar,
              email: user.email,
            };
        }

        throw new Error(MESSAGES.INVALID_CREDENTIALS);
      },
    }),
  ],
});
