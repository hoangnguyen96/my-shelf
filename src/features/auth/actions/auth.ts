"use server";

import { AuthError } from "next-auth";
import { MESSAGES } from "@app/constants";
import { User } from "@app/interface";
import { signIn, signOut } from "@app/auth";

export const authenticate = async (formData: Partial<User>) => {
  try {
    await signIn("credentials", formData);
  } catch (error) {
    if (error instanceof AuthError) {
      return MESSAGES.EMAIL_PASSWORD_INVALID;
    }
    throw error;
  }
};

export const logout = async () => {
  await signOut();
};
