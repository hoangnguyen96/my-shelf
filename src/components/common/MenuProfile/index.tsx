"use client";

import { ChevronDownIcon } from "@chakra-ui/icons";
import {
  Button,
  Flex,
  Menu,
  MenuButton,
  MenuItem,
  MenuList,
  Text,
} from "@chakra-ui/react";
import Avatar from "../Avatar";
import { logout } from "@app/actions/auth";
import { useRouter } from "next/navigation";
import { ROUTES } from "@app/constants";
import Link from "next/link";
import { useSession } from "next-auth/react";

const MenuProfile = () => {
  const router = useRouter();
  const { data: session } = useSession();

  const handleLogout = async () => {
    await logout();
    router.push(ROUTES.LOGIN);
  };

  return (
    <Menu>
      <MenuButton
        as={Button}
        rightIcon={<ChevronDownIcon color="dark.90" />}
        variant="outline"
        borderColor="borderDefault"
        bgColor="white"
        minW={205}
        h={50}
        borderRadius="33px"
        py="3px"
        pl="2px"
        pr="30px"
        boxShadow="0 0 4px -1px #a9a9a9"
      >
        <Flex justifyContent="flex-start" alignItems="center" gap={4}>
          <Avatar
            image={session?.user?.image || ""}
            width={45}
            height={45}
            border="2px solid white"
          />
          <Text size="xl" flex={1}>
            {session?.user?.name}
          </Text>
        </Flex>
      </MenuButton>
      <MenuList minW={205} borderRadius="10px" boxShadow="0 0 3px 0px #a9a9a9">
        <MenuItem>
          <Link href={ROUTES.PROFILE} style={{ width: "100%" }}>
            Profile
          </Link>
        </MenuItem>
        <MenuItem>
          <Link
            href={`${ROUTES.MY_BOOK_SHELF_FAVORITES}`}
            style={{ width: "100%" }}
          >
            Favorite
          </Link>
        </MenuItem>
        <MenuItem onClick={handleLogout}>
          <Text lineHeight="30px" w="100%">
            Logout
          </Text>
        </MenuItem>
      </MenuList>
    </Menu>
  );
};

export default MenuProfile;
