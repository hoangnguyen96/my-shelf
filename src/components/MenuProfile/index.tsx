"use client";

import { memo } from "react";
import { Session } from "next-auth";
import { ChevronDownIcon } from "@chakra-ui/icons";
import {
  Button,
  Flex,
  Menu,
  MenuButton,
  MenuItem,
  MenuList,
  Text,
} from "@chakra-ui/react";
import { useRouter } from "next/navigation";
import { ROUTES } from "@app/constants";
import { logout } from "@app/features/auth/actions";
import { Avatar } from "../common";

const MenuProfile = ({ session }: { session?: Session }) => {
  const router = useRouter();

  const handleLogout = async () => {
    await logout();
    router.push(ROUTES.LOGIN);
  };

  const handleRedirectProfile = () => router.push(ROUTES.PROFILE);

  const handleRedirectFavorites = () =>
    router.push(ROUTES.MY_BOOK_SHELF_FAVORITES);

  return (
    <Menu>
      <MenuButton
        as={Button}
        rightIcon={<ChevronDownIcon color="dark.90" />}
        variant="outline"
        borderColor="borderDefault"
        bgColor="white"
        minW={205}
        h={50}
        borderRadius="33px"
        py="3px"
        pl="2px"
        pr="30px"
        boxShadow="0 0 4px -1px #a9a9a9"
        justifyContent="space-between"
        sx={{ "& span": { display: "contents" } }}
      >
        <Flex
          justifyContent="flex-start"
          alignItems="center"
          gap={4}
          height="100%"
        >
          <Avatar
            image={session?.user?.image || ""}
            width={45}
            height={45}
            border="2px solid white"
          />
          <Text size="xl" flex={1}>
            {session?.user?.name || ""}
          </Text>
        </Flex>
      </MenuButton>
      <MenuList minW={205} borderRadius="10px" boxShadow="0 0 3px 0px #a9a9a9">
        <MenuItem onClick={handleRedirectProfile}>
          <Text lineHeight="30px" w="100%">
            Profile
          </Text>
        </MenuItem>
        <MenuItem onClick={handleRedirectFavorites}>
          <Text lineHeight="30px" w="100%">
            Favorites
          </Text>
        </MenuItem>
        <MenuItem onClick={handleLogout}>
          <Text lineHeight="30px" w="100%">
            Logout
          </Text>
        </MenuItem>
      </MenuList>
    </Menu>
  );
};

export default memo(MenuProfile);
