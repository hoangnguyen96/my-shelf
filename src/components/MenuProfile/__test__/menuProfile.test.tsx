import "@testing-library/jest-dom";
import { fireEvent, render, waitFor } from "@testing-library/react";
import { useSession } from "next-auth/react";
import { useRouter } from "next/navigation";
import { ROUTES } from "@app/constants";
import { logout } from "@app/features/auth/actions";
import MenuProfile from "..";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

jest.mock("@app/actions/auth", () => ({
  logout: jest.fn(),
}));

describe("MenuProfile", () => {
  const propsSession = {
    user: {
      isAdmin: true,
      email: "admin@gmail.com",
      id: "3733403",
      name: "admin",
      image: "https://i.ibb.co/RHMqQGr/man-1.png",
    },
    expires: "2024-12-31T23:59:59.999Z",
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Should render correctly snapshot", () => {
    expect(render(<MenuProfile session={propsSession} />)).toMatchSnapshot();
  });

  it("Should render correctly snapshot when data empty", () => {
    expect(render(<MenuProfile session={undefined} />)).toMatchSnapshot();
  });

  it("should call logout and redirect to login", async () => {
    const mockPush = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      push: mockPush,
    });

    const { getByText } = render(<MenuProfile session={propsSession} />);

    const logoutButton = getByText("Logout");
    fireEvent.click(logoutButton);

    await waitFor(() => {
      expect(logout).toHaveBeenCalled();
      expect(mockPush).toHaveBeenCalledWith(ROUTES.LOGIN);
    });
  });

  it("should redirect to profile", async () => {
    const mockPush = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      push: mockPush,
    });

    const { getByText } = render(<MenuProfile session={propsSession} />);

    const butRedirect = getByText("Profile");
    fireEvent.click(butRedirect);

    await waitFor(() => {
      expect(mockPush).toHaveBeenCalledWith(ROUTES.PROFILE);
    });
  });

  it("should redirect to profile", async () => {
    const mockPush = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      push: mockPush,
    });

    const { getByText } = render(<MenuProfile session={propsSession} />);

    const butRedirect = getByText("Favorites");
    fireEvent.click(butRedirect);

    // Use waitFor to ensure the async actions have been processed
    await waitFor(() => {
      expect(mockPush).toHaveBeenCalledWith(ROUTES.MY_BOOK_SHELF_FAVORITES);
    });
  });
});
