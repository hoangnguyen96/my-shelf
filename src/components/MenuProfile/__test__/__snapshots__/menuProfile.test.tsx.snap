// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuProfile Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-controls="menu-list-:r1:"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-button chakra-menu__menu-button css-115zmqy"
        id="menu-button-:r1:"
        type="button"
      >
        <span
          class="css-xl71ch"
        >
          <div
            class="css-yvf3fq"
          >
            <div
              class="css-bgqooo"
            >
              <img
                alt="Avatar user"
                data-nimg="fill"
                decoding="async"
                fetchpriority="high"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
              />
            </div>
            <p
              class="chakra-text css-1rr4qq7"
            >
              admin
            </p>
          </div>
        </span>
        <span
          class="chakra-button__icon css-1hzyiq5"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-1bho5id"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-1r8dsm8"
          id="menu-list-:r1:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8);"
          tabindex="-1"
        >
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="0"
            id="menu-list-:r1:-menuitem-:r2:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-60gkn2"
            >
              Profile
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="1"
            id="menu-list-:r1:-menuitem-:r3:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-60gkn2"
            >
              Favorites
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="2"
            id="menu-list-:r1:-menuitem-:r4:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-60gkn2"
            >
              Logout
            </p>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      aria-controls="menu-list-:r1:"
      aria-expanded="false"
      aria-haspopup="menu"
      class="chakra-button chakra-menu__menu-button css-115zmqy"
      id="menu-button-:r1:"
      type="button"
    >
      <span
        class="css-xl71ch"
      >
        <div
          class="css-yvf3fq"
        >
          <div
            class="css-bgqooo"
          >
            <img
              alt="Avatar user"
              data-nimg="fill"
              decoding="async"
              fetchpriority="high"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FRHMqQGr%2Fman-1.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <p
            class="chakra-text css-1rr4qq7"
          >
            admin
          </p>
        </div>
      </span>
      <span
        class="chakra-button__icon css-1hzyiq5"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-1bho5id"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <div
      class="css-0"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-1r8dsm8"
        id="menu-list-:r1:"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8);"
        tabindex="-1"
      >
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="0"
          id="menu-list-:r1:-menuitem-:r2:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Profile
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="1"
          id="menu-list-:r1:-menuitem-:r3:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Favorites
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="2"
          id="menu-list-:r1:-menuitem-:r4:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Logout
          </p>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MenuProfile Should render correctly snapshot when data empty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-controls="menu-list-:r6:"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-button chakra-menu__menu-button css-115zmqy"
        id="menu-button-:r6:"
        type="button"
      >
        <span
          class="css-xl71ch"
        >
          <div
            class="css-yvf3fq"
          >
            <div
              class="css-bgqooo"
            >
              <img
                alt="Avatar user"
                data-nimg="fill"
                decoding="async"
                fetchpriority="high"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
              />
            </div>
            <p
              class="chakra-text css-1rr4qq7"
            />
          </div>
        </span>
        <span
          class="chakra-button__icon css-1hzyiq5"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-1bho5id"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-1r8dsm8"
          id="menu-list-:r6:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8);"
          tabindex="-1"
        >
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="0"
            id="menu-list-:r6:-menuitem-:r7:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-60gkn2"
            >
              Profile
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="1"
            id="menu-list-:r6:-menuitem-:r8:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-60gkn2"
            >
              Favorites
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-1k756t9"
            data-index="2"
            id="menu-list-:r6:-menuitem-:r9:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-60gkn2"
            >
              Logout
            </p>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      aria-controls="menu-list-:r6:"
      aria-expanded="false"
      aria-haspopup="menu"
      class="chakra-button chakra-menu__menu-button css-115zmqy"
      id="menu-button-:r6:"
      type="button"
    >
      <span
        class="css-xl71ch"
      >
        <div
          class="css-yvf3fq"
        >
          <div
            class="css-bgqooo"
          >
            <img
              alt="Avatar user"
              data-nimg="fill"
              decoding="async"
              fetchpriority="high"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FSKHPQYq%2Favatar-default.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <p
            class="chakra-text css-1rr4qq7"
          />
        </div>
      </span>
      <span
        class="chakra-button__icon css-1hzyiq5"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-1bho5id"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <div
      class="css-0"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-1r8dsm8"
        id="menu-list-:r6:"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8);"
        tabindex="-1"
      >
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="0"
          id="menu-list-:r6:-menuitem-:r7:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Profile
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="1"
          id="menu-list-:r6:-menuitem-:r8:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Favorites
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-1k756t9"
          data-index="2"
          id="menu-list-:r6:-menuitem-:r9:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="chakra-text css-60gkn2"
          >
            Logout
          </p>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
